graph/src/data/store/mod.rs

```diff
@@ -647,6 +649,38 @@ pub trait TryIntoEntityIterator<E>: IntoIterator<Item = Result<(Word, Value), E>
 
 impl<E, T: IntoIterator<Item = Result<(Word, Value), E>>> TryIntoEntityIterator<E> for T {}
 
+#[derive(Debug, Error, PartialEq, Eq)]
+pub enum EntityValidationError {
```

Comment:
@mangas I added new `EntityValidationError` enum and migrated all the validation error to use it.